src/neuroglancer/webgl/context.ts

```diff
@@ -31,10 +31,7 @@ export function initializeWebGL(canvas: HTMLCanvasElement) {
     'antialias': false,
     'stencil': true,
   };
-  if (DEBUG_SHADERS) {
```

Comment:
I looked into this again and according to [StackOverflow](https://stackoverflow.com/questions/30628064/how-to-toggle-preservedrawingbuffer-in-three-js) it might be possible to avoid the performance drop by forcing a redraw right before you grab the screen... I would prefer this a lot over the `preserveDrawingBuffer` solution, because as you mentioned, it might have a big performance impact... Maybe you can call the `viewer.display.draw()` function before your `toDataURL()` call(?)